core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -253,9 +251,9 @@ public IndexShard(ShardId shardId, IndexSettings indexSettings, ShardPath path,
 
         this.engineConfig = newEngineConfig(translogConfig, cachingPolicy);
         this.suspendableRefContainer = new SuspendableRefContainer();
-        this.provider = provider;
         this.searcherWrapper = indexSearcherWrapper;
-        this.percolatorQueriesRegistry = new PercolatorQueriesRegistry(shardId, indexSettings, newQueryShardContext());
+        QueryShardContext queryShardContext = new QueryShardContext(idxSettings, provider.getClient(), indexCache.bitsetFilterCache(), indexFieldDataService, mapperService, similarityService, provider.getScriptService(), provider.getIndicesQueriesRegistry());
```

Comment:
This is the only place (outside of tests) that I can see where the QueryShardContext is not created using IndexService.newQueryShardContext(). I wonder f we should/can use the IndexService to create it here too?